### ISSUE TITLE
AbstractClassCanBeConcreteClass false negatives

### ISSUE THREAD

**Issue by FletchMcKee** [detekt/detekt#9155]
This was something I discovered while working on #9127. I noticed before that change, sometimes Signal-Android would report a code smell for this rule for DSLSettingsFragment and/or KeyboardEntryDialogFragment. 

To add to the confusion, there was an additional abstract ChangeMyStoryMembershipFragment that also shouldn't be abstract, but was never reported. After the 9127 change, it never reported for any of these fragments, despite all three violating this rule.

After some debugging I found that it wasn't reporting because `hasInheritedMember` was finding this inside Android's Fragment class and suppressing the report:
```java
private abstract static class OnPreAttachedListener {
  abstract void onPreAttached();
}
```

Looking through AbstractClassCanBeConcreteClass, I think there is a minor code change in `hasInheritedMember` that could fix this issue:
```diff
- (symbol as? KaClassSymbol)?.memberScope?.declarations.orEmpty().any {
+ (symbol as? KaClassSymbol)?.memberScope?.callables.orEmpty().any {
    it.modality == KaSymbolModality.ABSTRACT
}
```

I still don't know why before this KaLibraryModule change it would sometimes report two, sometimes one or none (maybe memory strain + StrongReferences?). It _seems_ like it is more deterministic after 9127 as it would always omit or always report after making the declarations/callables change, but it's hard to say when dealing with false negatives.

I can create a PR for this, but wanted to verify that this would be the expected behavior.

## Expected Behavior
An abstract class that only inherits non-callable classifiers (ex. abstract nested classes) from its supertype should be reported.

## Observed Behavior
The rule doesn't report this behavior. For example, all abstract classes that extend Android's Fragment class won't report due to the inner abstract `OnPreAttachedListener`.

## Steps to Reproduce
Adding this test case to AbstractClassCanBeConcreteClassSpec should allow it to pass, but it fails:
```kotlin
@Test
fun `reports when supertype only declares abstract nested class`() {
    val code = """
        abstract class Base {
            abstract class Nested
        }
        
        abstract class Derived : Base() {
            fun f() {}
        }
    """.trimIndent()
    val findings = subject.lintWithContext(env, code)
    assertThat(findings).singleElement()
        .hasMessage("An abstract class without an abstract member can be refactored to a concrete class.")
}
```
